File: app/src/main/java/com/ps/simplepapersoccer/Sound/FXPlayer.java
package com.ps.simplepapersoccer.Sound;

import android.app.Activity;
import android.content.Context;
import android.media.AudioManager;
import android.media.SoundPool;
import android.os.Build;

import com.ps.simplepapersoccer.R;

import java.util.HashMap;
import java.util.Map;

public class FXPlayer
{
    private SoundPool soundPool;
    private Map<Integer, Integer> soundMap;
    private AudioManager audioManager;

    public FXPlayer(Activity activity) {

        int maxStreams = 1;
        Context mContext = activity.getApplicationContext();
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
            soundPool = new SoundPool.Builder()
                    .setMaxStreams(maxStreams)
                    .build();
        } else {
            soundPool = new SoundPool(maxStreams, AudioManager.STREAM_MUSIC, 0);
        }

        soundMap = new HashMap<>();
        // fill your sounds
        soundMap.put(R.raw.bounce, soundPool.load(mContext, R.raw.bounce, 1));
        soundMap.put(R.raw.soccerkick, soundPool.load(mContext, R.raw.soccerkick, 2));
        soundMap.put(R.raw.failure, soundPool.load(mContext, R.raw.failure, 1));
        soundMap.put(R.raw.goodresult, soundPool.load(mContext, R.raw.goodresult, 1));

        audioManager = (AudioManager)mContext.getSystemService(Context.AUDIO_SERVICE);
    }

    public void playSound(int sound) {

        if (soundPool != null) soundPool.play(soundMap.get(sound), 1, 1, 1, 0, 1f);
    }

    public final void cleanUpIfEnd() {
        soundMap = null;
        soundPool.release();
        soundPool = null;
    }
}

File: app/src/main/java/com/ps/simplepapersoccer/Sound/FXPlayer.kt
package com.ps.simplepapersoccer.sound

import android.app.Activity
import android.content.Context
import android.media.AudioManager
import android.media.SoundPool
import android.os.Build
import android.util.SparseIntArray

import com.ps.simplepapersoccer.R

class FXPlayer(private val context: Context) {
    private var soundPool: SoundPool? = null
    private var soundMap: HashMap<Int, Int>
    private val audioManager: AudioManager

    init {

        val maxStreams = 1
        val soundPool = if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
            SoundPool.Builder()
                    .setMaxStreams(maxStreams)
                    .build()
        } else {
            SoundPool(maxStreams, AudioManager.STREAM_MUSIC, 0)
        }

        soundMap = HashMap()
        // fill your sounds
        soundMap[R.raw.bounce] = soundPool.load(context, R.raw.bounce, 1)
        soundMap[R.raw.soccerkick] = soundPool.load(context, R.raw.soccerkick, 2)
        soundMap[R.raw.failure] = soundPool.load(context, R.raw.failure, 1)
        soundMap[R.raw.goodresult] = soundPool.load(context, R.raw.goodresult, 1)

        audioManager = context.getSystemService(Context.AUDIO_SERVICE) as AudioManager

        this.soundPool = soundPool
    }

    fun playSound(sound: Int) {
        soundPool?.play(soundMap[sound]!!, 1f, 1f, 1, 0, 1f)
    }

    fun cleanUpIfEnd() {
        soundPool?.release()
        soundPool = null
    }
}
